voice/src/main/kotlin/handlers/UdpLifeCycleHandler.kt

```diff
@@ -4,45 +4,36 @@ package dev.kord.voice.handlers
 
 import dev.kord.common.annotation.KordVoice
 import dev.kord.voice.EncryptionMode
-import dev.kord.voice.gateway.*
-import dev.kord.voice.udp.AudioFramePollerConfigurationBuilder
-import dev.kord.voice.udp.DiscordUdpConnection
-import dev.kord.voice.udp.VoiceUdpConnection
+import dev.kord.voice.FrameInterceptorContextBuilder
+import dev.kord.voice.VoiceConnection
+import dev.kord.voice.gateway.Ready
+import dev.kord.voice.gateway.SelectProtocol
+import dev.kord.voice.gateway.SessionDescription
+import dev.kord.voice.gateway.VoiceEvent
+import dev.kord.voice.udp.AudioFrameSenderConfiguration
+import dev.kord.voice.udp.VoiceUdpConnectionConfiguration
 import io.ktor.util.network.*
-import kotlinx.coroutines.CoroutineDispatcher
-import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
 import mu.KotlinLogging
-import dev.kord.voice.gateway.VoiceEvent as VoiceEvent
+import kotlin.properties.Delegates
 
 private val udpLifeCycleLogger = KotlinLogging.logger { }
 
+@OptIn(KordVoice::class)
 internal class UdpLifeCycleHandler(
     flow: Flow<VoiceEvent>,
-    private val send: suspend (Command) -> Unit,
-    private val poll: (AudioFramePollerConfigurationBuilder) -> Unit,
-    private val udpDispatcher: CoroutineDispatcher = Dispatchers.Default
+    private val connection: VoiceConnection
 ) : EventHandler<VoiceEvent>(flow, "UdpInterceptor") {
-    private var framePollerConfigurationBuilder = AudioFramePollerConfigurationBuilder()
-    private var udp: DiscordUdpConnection? = null
+    private var ssrc by Delegates.notNull<UInt>()
```

Comment:
This is not thread safe, there is no happens-before relationship between setting and using `ssrc`.
-> `@Volatile` or `atomic`?